### DISCUSSION TITLE
Persistence user on rabbitmq

### DISCUSSION THREAD

**Discussion by ghost** [docker-library/rabbitmq#607] [Q&A]
So I was make my rabbitmq and I noticed that, when I persiste the volume, my users don't save when I make docker-compose down  and up, and I saw that, my rabbitmq creates a new "cluester" all time that I make the docker compose up, so how can I stay with a one cluster, have any thing that can I to do to pin just one cluster name, to use one cluster with the save config 
![image](https://user-images.githubusercontent.com/88913232/217075909-c7d50475-312f-4234-9081-6433c8853493.png)

**Comment by michaelklishin** [comment]
@YukioBunny you already asked exactly the same question about node's data directory persistence in #605 and was offered recommendations. Nodes only initialize new clusters if they do not have an existing data directory to use (because the volume is not persistent or the data directory path has changed).

This will involve log messages such as

```
2023-02-06 15:56:40.145883-05:00 [info] <0.229.0> Node database directory at /tmp/123 is empty. Assuming we need to join an existing cluster or initialise from scratch...
```

```
2023-02-06 15:56:40.221024-05:00 [info] <0.229.0> Initialising internal cluster ID to 'rabbitmq-cluster-id-IDN4yCLL1zZINr2hifW8pw'
```

in node logs.

And then they [perform peer discovery](https://rabbitmq.com/cluster-formation.html#peer-discovery-how-does-it-work) on the first boot, and log the result.

**Comment by yosifkit** [comment]
You likely need to provide a consistent `hostname`: https://github.com/docker-library/docs/tree/bb6b29b9a31fe8fe9e083c989b3091914e301fd6/rabbitmq#running-the-daemon
